python/cudf/dataframe/dataframe.py

```diff
@@ -2361,28 +2361,82 @@ def quantile(self,
                                          quant_index=False)
         return result
 
-    def select_dtypes(self, include=None):
+    def select_dtypes(self, include=None, exclude=None):
         """Return a subset of the DataFrame’s columns based on the column dtypes.
 
         Parameters
         ----------
         include : str or list
             which columns to include based on dtypes
+        exclude : str or list
+            which columns to exclude based on dtypes
 
         """
 
+        # code modified from:
+        # https://github.com/pandas-dev/pandas/blob/master/pandas/core/frame.py#L3196
+
         if not isinstance(include, (list, tuple)):
-            include = [include]
+            include = (include,) if include is not None else ()
+        if not isinstance(exclude, (list, tuple)):
+            exclude = (exclude,) if exclude is not None else ()
+
         df = DataFrame()
 
-        include = [pd.core.dtypes.common.pandas_dtype(d) for d in include]
+        # infer_dtype_from_object can distringuish between
+        # np.float and np.number
+        selection = tuple(map(frozenset, (include, exclude)))
+        include, exclude = map(
+            lambda x: frozenset(
+                map(pd.core.dtypes.common._get_dtype_from_object, x)),
```

Comment:
Is there a non-internal function we could use here? Such as `pandas.api.types.infer_dtype`?

Reply:
We can't use `infer_dtype` here because we are trying to map valid inputs for include/exclude to a dtype.  `infer_dtype` is used to infer the type of the passed value

Reply:
in 0.24 `_get_dtype_from_object` is replaced with `infer_dtype_from_object`

Reply:
This function doesn't exist in 0.24.1 though as well, hmm.

Reply:
I'll spend some time looking around and post back

Reply:
@kkraus14 at the beginning of the year the name of the function changed but it is largely the same: https://github.com/pandas-dev/pandas/commit/19f715c51d16995fc6cd0c102fdba2f213a83a0f#diff-f96467bea27cd795c0e4fdb4c70bb764

Reply:
What do you think about trying to import `infer_dtype_from_object` and failing over to `_get_dtype_from_object` as  `infer_dtype_from_object` ?

Reply:
👍